play-frontend-hmrc-play-30/src/main/scala/uk/gov/hmrc/hmrcfrontend/views/viewmodels/header/v2/HeaderParams.scala

```diff
@@ -0,0 +1,126 @@
+
+  implicit def headerToHeaderParams(header: Header): HeaderParams =
+    HeaderParams(
+      headerUrls = HeaderUrls(
+        homepageUrl = header.homepageUrl,
+        serviceUrl = header.serviceUrl,
+        signOutHref = header.signOutHref,
+        assetsPath = header.assetsPath
+      ),
+      headerNames = HeaderNames(
+        productName = header.productName,
+        serviceName = header.serviceName
+      ),
+      headerTemplateOverrides = HeaderTemplateOverrides(
+        containerClasses = header.containerClasses,
+        classes = header.classes,
+        attributes = header.attributes
+      ),
+      navigationOverrides = NavigationOverrides(
+        navigation = header.navigation,
+        navigationClasses = header.navigationClasses,
+        navigationLabel = header.navigationLabel
+      ),
+      banners = Banners(
+        displayHmrcBanner = header.displayHmrcBanner,
+        phaseBanner = header.phaseBanner,
+        userResearchBanner = header.userResearchBanner,
+        additionalBannersBlock = header.additionalBannersBlock
+      ),
+      menuButtonOverrides = MenuButtonOverrides(
+        menuButtonLabel = header.menuButtonLabel,
+        menuButtonText = header.menuButtonText
+      ),
+      logoOverrides = LogoOverrides(
+        useTudorCrown = header.useTudorCrown,
+        rebrand = header.rebrand
+      ),
+      language = header.language,
+      inputLanguageToggle = header.languageToggle
+    )
+
+  implicit def headerParamsToHeader(headerParams: HeaderParams): Header =
+    Header(
+      homepageUrl = headerParams.headerUrls.homepageUrl,
+      assetsPath = headerParams.headerUrls.assetsPath,
+      productName = headerParams.headerNames.productName,
+      serviceName = headerParams.headerNames.serviceName,
+      serviceUrl = headerParams.headerUrls.serviceUrl,
+      navigation = headerParams.navigationOverrides.navigation,
+      navigationClasses = headerParams.navigationOverrides.navigationClasses,
+      containerClasses = headerParams.headerTemplateOverrides.containerClasses,
+      classes = headerParams.headerTemplateOverrides.classes,
+      attributes = headerParams.headerTemplateOverrides.attributes,
+      language = headerParams.language,
+      displayHmrcBanner = headerParams.banners.displayHmrcBanner,
+      useTudorCrown = headerParams.logoOverrides.useTudorCrown,
+      signOutHref = headerParams.headerUrls.signOutHref,
+      inputLanguageToggle = headerParams.languageToggle,
+      userResearchBanner = headerParams.banners.userResearchBanner,
+      phaseBanner = headerParams.banners.phaseBanner,
+      additionalBannersBlock = headerParams.banners.additionalBannersBlock,
+      menuButtonLabel = headerParams.menuButtonOverrides.menuButtonLabel,
+      menuButtonText = headerParams.menuButtonOverrides.menuButtonText,
+      navigationLabel = headerParams.navigationOverrides.navigationLabel,
+      rebrand = headerParams.logoOverrides.rebrand
+    )
```

Comment:
do we need to have it both ways, or just Header => HeaderParams - with it going this way I think we'll have the chance to lose some data when we add new fields to HeaderParams